### DISCUSSION TITLE
Send X-Autoreply To Message Route

### DISCUSSION THREAD

**Discussion by danpastori** [postalserver/postal#1484] [Feature suggestions]
Hi!

We are handling a bunch of email scenarios and would like to catch any emails that come back from an Out Of Office auto reply. In the header from the auto reply email, there is an `x-autoreply` header:
<img width="216" alt="Screen Shot 2019-06-12 at 5 14 47 PM" src="https://user-images.githubusercontent.com/2365103/59394795-eec23980-8d35-11e9-99d4-0f538730fa91.png">

Is it on the map to send this to the message route as JSON? Would a PR help, I could try to add this and submit it if it's of interest?

Thanks!

**Comment by willpower232** [comment]
I'm not sure if this is on atechs plans but it certainly sounds like a nice idea. However judging by a bit of light internet research detecting an autoreply message accurately can be a challenge.

https://github.com/BOOKATIGER/is-autoreply/blob/master/index.js
https://github.com/jpmckinney/multi_mail/wiki/Detecting-autoresponders
https://www.jitbit.com/maxblog/18-detecting-outlook-autoreplyout-of-office-emails-and-x-auto-response-suppress-header/

That said, there may be room for more generic routing based on you specifying headers depending on your use case. I.E. if you're reasonably sure of the headers in messages that Postal will receive.

**Comment by danpastori** [comment]
@willpower232 Thanks for the quick reply! During our testing we noticed that it's super hard to determine an auto reply, but Postal picks up the `x-autoreply`. I was wondering if when that header is present, we could pass a flag to the web route with the message data? An example from the JSON in the callback received by the server would look like this: 

```
{
  "id": 255,
  "rcpt_to": "",
  "mail_from": "",
  "token": "",
  "subject": "Out of Office Re: Test OOO #1",
  "message_id": "",
  "timestamp": ,
  "size": "",
  "spam_status": "NotSpam",
  "bounce": false,
  "received_with_ssl": true,
  "to": "",
  "cc": null,
  "from": "",
  "date": "",
  "in_reply_to": null,
  "references": null,
  "html_body": "",
  "attachment_quantity": 0,
  "plain_body": null,
  "replies_from_plain_body": null,
  "attachments": [],
  "auto_reply": true // This would be attached if the auto reply is detected.
}
```

**Comment by willpower232** [comment]
As I said it would be difficult to make such a parameter accurate for every situation.

You could also choose to receive the raw message rather than the JSON alternative which I hope includes all the headers?
